service/history/historyEngine.go

```diff
@@ -2069,9 +2080,14 @@ func (e *historyEngineImpl) RequestCancelWorkflowExecution(
 	}
 
 	return workflow.UpdateCurrentWithActionFunc(ctx, e.executionCache, e.executionManager, domainID, workflowExecution, e.timeSource.Now(),
-		func(wfContext execution.Context, mutableState execution.MutableState) (*workflow.UpdateAction, error) {
+		func(ctx context.Context, wfContext execution.Context, mutableState execution.MutableState) (*workflow.UpdateAction, error) {
 			if !mutableState.IsWorkflowExecutionRunning() {
-				return nil, workflow.ErrAlreadyCompleted
+				return nil, execution.VersionBasedError(
+					workflow.ErrAlreadyCompleted,
+					"1.7.0",
+					yarpc.CallFromContext(ctx).Header(common.FeatureVersionHeaderName),
+					workflow.ErrNotExists,
```

Comment:
I am not entirely sure about this, does the call from frontend to history contains client version in the header?